download.py

```diff
@@ -848,11 +839,13 @@ def download_post(task: client_tasks.Task):
     # don't do this stuff in editmode and other modes, just wait...
     # we don't remove the task before it's actually possible to remove it.
     if bpy.context.mode != "OBJECT" and (at == "model" or at == "material"):
-        return done
+        # try to switch to object mode - if it's not possible, propagate exception higher up
+        bpy.ops.object.mode_set(mode="OBJECT")
```

Comment:
Pred timhle PRkem to proste vzdalo. Pritom jsme v timeru a vlastne to klidne muzeme zmenit. Takze tu volam mode_set, abysme to nevzdavali predcasne - ono je mozny, ze proste:
1. jsem v edit
2. dam download a ten me prepne do object
3. prepnu se do edit, protoze chci
4. za dlouho dobehne velky download a append fail...

Pritom proste jen muzeme prepnout do OBJECT mode a s klidnem ten asset pripnout.

Je to v pohode uvaha? @vilemduha

Reply:
Tohle ted mergenu, protoze to je daleko lepsi nez rozbity. Vytvorim tady ale issue, aby se ty tasky postponovaly, odkladaly, tak jako to bylo driv, protoze to je lepsi, rusi to min v praci.